aws-cost-exporter.py

```diff
@@ -7,96 +7,98 @@
-	)
-
-	#Create an empty dictionary
-	projectValues = {}
-
-	#Run the response and make a dictionary with tag name and tag value
-	for project in response["ResultsByTime"][0]["Groups"]:
 
-		#Search for tag
-		namestring = project['Keys'][0]
-		name = re.search("\$(.*)", namestring).group(1)
-
-		#If name is none, let's defined it as Other
-		if name is None or name == "":
-			name = "Other"
-
-		#Get the value
-		amount = project['Metrics']['BlendedCost']['Amount']
-		#Format the time to 0.2 points
-		amount = "{0:.2f}".format(float(amount))
+def getCosts():
+    # Create a boto3 connection with cost explorer
+    client = boto3.client('ce')
+
+    # Get the current time
+    now = datetime.datetime.utcnow()
+
+    # Set the end of the range to start of the current day
+    end = datetime.datetime(
+        year=now.year, month=now.month, day=now.day, hour=now.hour)
+
+    # Subtract a day to define the start of the range
+    start =  datetime.datetime(
+        year=now.year, month=now.month, day=now.day, hour=0)
+
+    # Convert them to strings
+    start = start.strftime('%Y-%m-%dT%H:00:00Z')
+    end = end.strftime('%Y-%m-%dT%H:00:00Z')
+
+    print("Starting script searching by the follow time range")
+    print(start + " - " + end)
+
+    # Call AWS API to get costs
+    response = client.get_cost_and_usage(
+        TimePeriod={
+            'Start': start,
+            'End':  end
+        },
+        Granularity='HOURLY',
+        Metrics=['BlendedCost'],
+        GroupBy=[
+            {
+                'Type': 'TAG',
+                'Key': tagProject
+            }
+        ]
+    )
+
+    # Create an empty dictionary
+    projectValues = []
+
+    # Run the response and make a dictionary with tag name and tag value
+    for project in response["ResultsByTime"]:
+        
+        # Get the value
+        group = project["Groups"]
 
-		#Append the values in the directionary
-		projectValues[name] = float(amount)
+        if not group:
+            continue
+        amount = group[0]['Metrics']['BlendedCost']['Amount']
+        
+        # Format the time to 0.2 points
+        amount = "{0:.2f}".format(float(amount))
 
-	#Return the dictionary with all those values
-	return projectValues
+        # Append the values in the directionary
+        projectValues.append(float(amount))
 
+    # Return the dictionary with all those values
+    return projectValues
 
-#Start classe collector
+# Start classe collector
 class costExporter(object):
 
-	def collect(self):
+    def collect(self):
+
+        # Expose the metric
+        # Create header
+        metric = Metric('aws_project_cost',
+                        'Total amount of costs for project', 'gauge')
 
-		#Expose the metric
-		#Create header
-		metric = Metric('aws_project_cost','Total amount of costs for project','gauge')
+        # Run the retuned dictionary and expose the metrics
+        for cost in getCosts():
+            metric.add_sample('aws_project_cost', value=cost,
```

Comment:
Do you have a sample output of this metrics endpoint? I think you aren't exposing the `Name` tag so it will only expose a total value, not a value by tag `Name`. I'm right?

Reply:
Yes! You are right!

We are currently getting all resources that have tag key `Name` but the AWS API does not bring us the value of the tag. That is, we are just handling values of aws costs broken per hour, as you said. Is this a real problem? If so, I have to think in another way to get this results, maybe filtering by resource type instead of all resources at same time, I dont know.

Reply:
e.g: 

![image](https://user-images.githubusercontent.com/22901121/91089099-76d5eb80-e629-11ea-9709-287cd66426b0.png)

Reply:
I had to change some parts of code to handle in a different way from the original one. Changed the time period, groupBy, granularity and the response to deal with all changes

Reply:
Yep, sadly it's a problem. We need splitted values (preferably by `Product`, but with `Name` we can do some regex), it's the only way to group with values of `kube-cost`. Remember what we wrote in the issue:
```
Visão por produto somando Serviços AWS
Visão por APP somando Serviços AWS
```

We only can do that splitting those values.

Reply:
Yes, I remember. But I remember that we discussed to keep only tag `Name` for now, too. But I will change the logic to split by this tags. Thanks

Reply:
Yep as I said `Name` maybe will work if we use some regex. The main problem is, currently isn't splitting by anything, at least by `Name` will be necessary.

Reply:
@phspagiari I've refactored almost all code to attend our need. It's now splitting by `app` and `product`. Look below how the metrics will be available to prometheus scrape:

![image](https://user-images.githubusercontent.com/22901121/91671127-2f080600-eafa-11ea-9fa6-6dfed4521834.png)

I think now it's good. If no, let me know.